### PR TITLE
Fixes #29940 - add select to FormField

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Add select and optgroupselect to be renderable within FormField

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/common/forms/FormField.fixtures.js [hunk 1]
```diff
@@ -15,6 +15,20 @@ export const dateTimeWithErrorProps = {
...
+export const selectProps = {
+  type: 'select',
+  id: 'date-time',
+  label: 'Grouped select',
+  name: 'group[groupedSelect]',
+  options: [
+    { true: 'Ungrouped true' },
+    { groupLabel: 'Group1', children: { rnd: 'Group1 opt1' }},
+    { false: 'Ungrouped false' },
+    { groupLabel: 'Group2', children: [['rnd2', 'Group2 opt2']]},
```
Comment (xprazak2) [inline webpack/assets/javascripts/react_app/components/common/forms/FormField.fixtures.js]: Should the label in the Group2 child be 'Group2 opt2'?

**Comment by xprazak2** [comment]
Snapshots need a refresh.

**Comment by ezr-ondrej** [comment]
Travis is happy now and I've fixed the fixtures and kept only two options style (eighter `[{value: 'a', label: 'b'}]` or `{a: 'b'}`)

**Review by xprazak2 — APPROVED** [review]

**Comment by xprazak2** [comment]
Thanks!

**Pull request merged**